File: sdks/ruby/lib/novabilling/types/checkout_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class CheckoutResponse < Internal::Types::Model
      field :checkout_url, -> { String }, optional: false, nullable: false, api_name: "checkoutUrl"
      field :payment_id, -> { String }, optional: false, nullable: false, api_name: "paymentId"
      field :provider, -> { String }, optional: false, nullable: false
      field :expires_at, -> { String }, optional: false, nullable: false, api_name: "expiresAt"
    end
  end
end

File: sdks/csharp/src/NovaBilling/Types/CheckoutResponse.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record CheckoutResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("checkoutUrl")]
    public required string CheckoutUrl { get; set; }

    [JsonPropertyName("paymentId")]
    public required string PaymentId { get; set; }

    [JsonPropertyName("provider")]
    public required string Provider { get; set; }

    [JsonPropertyName("expiresAt")]
    public required string ExpiresAt { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
